hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/pipeline/TestPipelineManagerMXBean.java

```diff
@@ -66,29 +65,35 @@ public void init()
   public void testPipelineInfo() throws Exception {
     ObjectName bean = new ObjectName(
         "Hadoop:service=SCMPipelineManager,name=SCMPipelineManagerInfo");
+    Map<String, Integer> pipelineStateCount = cluster
+        .getStorageContainerManager().getPipelineManager().getPipelineInfo();
 
-    TabularData data = (TabularData) mbs.getAttribute(bean, "PipelineInfo");
-    Map<String, Integer> datanodeInfo = cluster.getStorageContainerManager()
-        .getPipelineManager().getPipelineInfo();
-    verifyEquals(data, datanodeInfo);
+    GenericTestUtils.waitFor(() -> {
+      try {
+        final TabularData data = (TabularData) mbs.getAttribute(
+            bean, "PipelineInfo");
+        for (String state : pipelineStateCount.keySet()) {
+          final Integer count = pipelineStateCount.get(state);
```

Comment:
```
M P WMI: org.apache.hadoop.ozone.scm.pipeline.TestPipelineManagerMXBean.lambda$testPipelineInfo$0(ObjectName, Map) makes inefficient use of keySet iterator instead of entrySet iterator  At TestPipelineManagerMXBean.java:[line 76]
```

https://github.com/devmadhuu/ozone/actions/runs/7020325124/job/19099946501#step:7:11